mitodl/mit-open
# refactor profile and onboarding

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4860
Closes https://github.com/mitodl/hq/issues/4804

### Description (What does it do?)
This PR refactors `OnboardingPage` and `ProfileEditForm` to use the `formik` library for form management. As part of this process, a number of other changes were made:

 - Form controls swapped out for more generic components from `ol-components`
 - Various spacings, styles and copy updated to match the designs
 - Learning format updated to be multiple choice

### Screenshots (if appropriate):
![image](https://github.com/user-attachments/assets/3a57a8ba-471c-4e68-8c81-6c526b6d0a67)
![image](https://github.com/user-attachments/assets/5458d8a0-08de-41fd-88be-8444acb971b9)
![image](https://github.com/user-attachments/assets/361360b7-3d77-4f20-8774-4997ddef0a20)
![image](https://github.com/user-attachments/assets/fecf2a25-b2a5-4d20-89b1-65d8c91d627f)
![image](https://github.com/user-attachments/assets/a94d8a15-8c39-4cfb-b88f-db7afd48fe4a)

![image](https://github.com/user-attachments/assets/1946a337-7800-4a26-a749-e2c78faa6a9a)
![image](https://github.com/user-attachments/assets/f98b37ff-29bc-4c51-8ca4-ea90968e6632)
![image](https://github.com/user-attachments/assets/ee5697c7-0250-4a4a-84be-b77490d71a49)
![image](https://github.com/user-attachments/assets/96b8a5e5-a0e5-48df-959c-15f00b02617a)
![image](https://github.com/user-attachments/assets/29333b1e-b2ad-44ac-afcd-e5593b6b9fde)

![image](https://github.com/user-attachments/assets/b4d83525-ff79-4d12-b739-9d2afca89213)
![image](https://github.com/user-attachments/assets/57eb8be9-3265-4f14-b2cd-68d19feb6cd7)

### How can this be tested?
 - Spin up this branch of `mit-open`
 - Visit http://localhost:8063/onboarding
 - Follow the onboarding flow and ensure that the styles and copy match the designs
 - Visit http://localhost:8063/dashboard/#profile
 - Ensure that the styles and copy match the designs
 - Change some info on your profile and verify that the "Update" button becomes available
 - Save your profile, refresh the page and make sure the changes have persisted

### Additional Context
You may notice that the `Radio` styles don't exactly match the designs. Between this and `Checkbox`, some more extensive refactoring needs to be done that is outside the scope of this PR.
